File: Algorithms/Binary Search/binary_search.cpp
//Binary Search Algorithm in c++
#include<iostream>
#include<stdlib.h>
using namespace std;

//Binary Search Function
int binary_search(int* array, int key, int low, int high)
{
    int mid;
    mid =(int)((low+high)/2);
    if (low == high)
    {
        if(array[low] == key)
        {
            return low;
        }
        else
        {
            return -1;
        }
    }
    else if(array[mid] == key)
    {
        return mid;
    }
    else if(array[mid] > key)
    {
        return binary_search(array, key, low, mid-1);
    }
    else
    {
        return binary_search(array, key, mid+1, high);
    }
}

//Main Function
int main()
{
    int length,element;
    //getting array size from user
    cout<<"Enter the size of array : ";
    cin>>length;
    //empty array creation
    int* array;
    array = (int*)malloc(length*sizeof(int));
    //loop to insert the array items into array
    for(int i=0;i<length;i++)
    {
        cout<<"Enter item no."<<i+1<<" in the array : ";
        cin>>element;
        array[i] = element;
    }
    int key;
    cout<<"Enter the element to search : ";
    cin>>key;
    //calling the function
    int result;
    int low = 0;
    int high = length-1;
    result = binary_search(array, key, low, high);
    //printing the result
    if (result>=0)
    {
        cout<<"The Element is found in position "<<result;
    }
    else
    {
        cout<<"The Element is not in the array";
    }
    return 0;
}

File: Algorithms/Binary Search/binary_search.py
#Binary Search Algorithm in python
#Binary Search Function
def binary_search(array,key,low,high):
    mid = int((low+high)/2)
    #comparision
    if low == high :
        if array[low] == key:
            return low
        else:
            return -1
    elif array[mid] == key:
        return mid
    elif array[mid] > key:
        return binary_search(array,key,low,mid-1)
    else:
        return binary_search(array,key,mid+1,high)
    
#Main function
#empty array creation
array = []
#getting array size from user
length = int(input("Enter the size of array : "))
#loop to append the array items into array
for i in range (length):
    array_item = int(input(f"Enter the {i+1}th item in the array : "))
    array.append(array_item)
key = int(input("Enter the element to search : "))
low = 0
high = length - 1
#calling the function
pos = binary_search(array,key,low,high)
#printing the result
if pos>=0:
    print(f"The element is found in {pos}th position")
else:
    print("Element is not in the array")